tests/python/gaia-ui-tests/gaiatest/apps/wallpaper/app.py

```diff
@@ -14,8 +14,9 @@ class Wallpaper(Base):
     _stock_wallpapers_locator = (By.CLASS_NAME, 'wallpaper')
 
     def tap_wallpaper_by_index(self, index):
-        self.wait_for_condition(lambda m: len(m.find_elements(*self._stock_wallpapers_locator)) >= index,
-            message = '%s Wallpapers not present after timeout' % index)
+        Wait(self.marionette).until(
+            lambda m: len(m.find_elements(*self._stock_wallpapers_locator)) >= (index + 1),
+            message='%d wallpaper(s) not present after timeout' % (index + 1))
```

Comment:
Because index is zero based and here we're waiting for a count of elements. I've slipped in a fix, but can move it out if preferred.

Dave Hunt

> On 27 Mar 2015, at 14:43, Martijn <notifications@github.com> wrote:
> 
> In tests/python/gaia-ui-tests/gaiatest/apps/wallpaper/app.py:
> 
> > @@ -14,8 +14,9 @@ class Wallpaper(Base):
> >      _stock_wallpapers_locator = (By.CLASS_NAME, 'wallpaper')
> >  
> >      def tap_wallpaper_by_index(self, index):
> > -        self.wait_for_condition(lambda m: len(m.find_elements(*self._stock_wallpapers_locator)) >= index,
> > -            message = '%s Wallpapers not present after timeout' % index)
> > +        Wait(self.marionette).until(
> > +            lambda m: len(m.find_elements(*self._stock_wallpapers_locator)) >= (index + 1),
> > +            message='%d wallpaper(s) not present after timeout' % (index + 1))
> The original had ' >= index' and you're replacing it with >= (index + 1). Why is that?
> It seems to work though with the test that's using tap_wallpaper_by_index (test_settings_wallpaper.py).
> 
> —
> Reply to this email directly or view it on GitHub.
>